### PR TITLE
openmvg: add with_jpeg option

### PR DESCRIPTION
**Pull request by maksim-petukhov**
Specify library name and version:  **openmvg/\***

The option allows to use libjpeg-turbo instead of libjpeg

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by valgur — APPROVED** [review]

File: recipes/openmvg/all/conanfile.py [hunk 1]
```diff
@@ -30,13 +30,15 @@ class Openmvgconan(ConanFile):
         "with_openmp": [True, False],
         "with_avx": [False, "avx", "avx2"],
         "programs": [True, False],
+        "jpeg": ["libjpeg", "libjpeg-turbo"]
```
Comment (SpaceIm) [inline recipes/openmvg/all/conanfile.py:33]: Could you change this option name to be consistent with other CCI recipes please? There is already enough inconsistencies in CCI ;)

Also you can add mozjpeg as other CCI recipes already do.

**Comment by valgur** [comment]
Could the default be switched to libjpeg-turbo as well, perhaps?

File: recipes/openmvg/all/conanfile.py (hunk 1 above)
Comment (maksim-petukhov) [reply recipes/openmvg/all/conanfile.py]: Done

**Comment by maksim-petukhov** [comment]
@valgur I don't think so. Other recipes have `libjpeg` as a default, so we shouldn't use `libjpeg-turbo` as a default there because it might result in a conflict.

File: recipes/openmvg/all/conanfile.py (hunk 1 above)
Comment (maksim-petukhov) [reply recipes/openmvg/all/conanfile.py]: `"with_libjpeg":` - 10 recipes
`"with_jpeg":` - 21 recipes
`"jpeg":` - 5 recipes
Therefore I stopped on `with_jpeg`

File: recipes/openmvg/all/conanfile.py [hunk 2]
```diff
@@ -66,7 +68,12 @@ def requirements(self):
         self.requires("coin-utils/2.11.6")
         self.requires("eigen/3.4.0", transitive_headers=True)
         self.requires("flann/1.9.2", transitive_headers=True, transitive_libs=True)
-        self.requires("libjpeg/9e")
+        if self.options.with_jpeg == "libjpeg":
+            self.requires("libjpeg/9e")
+        elif self.options.with_jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.1.5")
```
Comment (valgur) [inline recipes/openmvg/all/conanfile.py:74]:
```suggestion
            self.requires("libjpeg-turbo/3.0.0")
```
Can probably bump the version soon after #18417 is merged.

**Review by ericLemanissier — APPROVED** [review]

**Comment by maksim-petukhov** [comment]
v2 pipeline is waiting for #17436

**Review by jcar87 — APPROVED** [review]

**Comment by maksim-petukhov** [comment]
@jcar87 There is a #17436 PR from @SpaceIm and I think your changes make it unnecessary. Should I mention #17436 in initial PR message?

**Review by SSE4 — APPROVED** [review]

**Pull request merged**